### PR TITLE
Task "Js pre oop"

### PR DESCRIPTION
**Pull request by DmitryHniezdilov**
### A Tiny JS World

[Demo](https://dmitryhniezdilov.github.io/task-a-tiny-JS-world/) |
[Code base](https://github.com/DmitryHniezdilov/task-a-tiny-JS-world/tree/gh-pages)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  saying: "A career is born in public – talent in privacy.",
+};
+
+const listOfInhabitants = [dog, cat, bear, men, women];
+
+// ======== OUTPUT ========
+
+listOfInhabitants.forEach(
+  ({ species, name, gender, legs, hands, saying } = item) => {
+    print(`${species}; ${name}; ${gender}; ${legs}; ${hands}; ${saying}`);
```
Comment (artem-trubin) [inline submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js:61]: There is a pattern here. A repeated action. Imagine a situation where you would need to change `;` with some other separator, it wouldn't be quick here if there would be more properties. Find a way to specify properties to print and then automate the output.

File: submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const bear = {
+  species: "character",
+  name: "Winnie",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Think, think, think!",
+};
+
+const men = {
```
Comment (artem-trubin) [inline submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js:37]: `men` and `women` are plurals. Your objects describe a single man and a single woman.

**Review by artem-trubin — CHANGES_REQUESTED** [review]
Hi, nice work, let's improve it a little.

File: submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js (hunk 2 above)
Comment (DmitryHniezdilov) [reply submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js]: THX for the correction. I'm confused 😕

File: submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js (hunk 1 above)
Comment (DmitryHniezdilov) [reply submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js]: THX for the explanation. I made the same automated output. Done

File: submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  saying: "A career is born in public – talent in privacy.",
+};
+
+const listOfInhabitants = [dog, cat, bear, man, woman];
+
+// ======== OUTPUT ========
+
+listOfInhabitants.forEach((item) => {
+  const separator = "; ";
+  const listOfValues = Object.values(item).join(separator);
```
Comment (artem-trubin) [inline submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js:61]: Okay you're almost there. Task specifies not to use methods like `Object.values` and similar to them. Using them, you cannot rely on properties' order and presence. You can specify properties by hand, though.

**Review by artem-trubin — CHANGES_REQUESTED** [review]

File: submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js (hunk 3 above)
Comment (DmitryHniezdilov) [reply submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js]: THX for the correction. I hope that it's ok.

Comment (DmitryHniezdilov) [reply submissions/DmitryHniezdilov/task-a-tiny-JS-world/index.js]: This code I updated only in the work branch in the current repo :
https://github.com/DmitryHniezdilov/task-a-tiny-JS-world/blob/task-a-tiny-JS-world/index.js.

**Review by artem-trubin — APPROVED** [review]
Nice idea! Clever use of reduce.

**Comment by DmitryHniezdilov** [comment]
Thank you.

**Pull request merged**